server/coin-scraper.js

```diff
@@ -9,25 +9,30 @@ module.exports = class CoinScraper {
     this.previousItemsMap = {};
   }
 
-  fetchMostRecentListedItems() {
-    const requestOptions = {
-      method: 'GET',
-      uri: 'https://pro-api.coinmarketcap.com/v1/cryptocurrency/listings/latest',
-      qs: {
-        'start': '1',
-        'limit': '200',
-        'sort': 'date_added',
-        'sort_dir': 'asc',
-        'aux': 'date_added'
-      },
-      headers: {
-        'X-CMC_PRO_API_KEY': CMC_API_KEY
-      },
-      json: true,
-      gzip: true
-    };
-
-    return rp(requestOptions);
+  async fetchMostRecentListedItems() {
+    try {
+      const requestOptions = {
+        method: 'GET',
+        url: 'https://pro-api.coinmarketcap.com/v1/cryptocurrency/listings/latest',
+        qs: {
+          'start': '1',
+          'limit': '200',
+          'sort': 'date_added',
+          'sort_dir': 'asc',
+          'aux': 'date_added'
+        },
+        headers: {
+          'X-CMC_PRO_API_KEY': CMC_API_KEY
+        },
+        json: true,
+        gzip: true
+      };
+      const response = await axios(requestOptions);
```

Comment:
@Shapokhan the data is not coming in in the (date) sorted order (i-e most recent first) that is expected: here is a dump of the first few items in the returned array:
```
{
    id: 1,
    name: 'Bitcoin',
    symbol: 'BTC',
    slug: 'bitcoin',
    num_market_pairs: 9551,
    date_added: '2013-04-28T00:00:00.000Z',
    tags: [
      'mineable',
      'pow',
      'paradigm-xzy-screener'
    ],
    max_supply: 21000000,
    circulating_supply: 18713050,
    total_supply: 18713050,
    platform: null,
    cmc_rank: 1,
    last_updated: '2021-05-18T15:19:02.000Z',
    quote: { USD: [Object] }
  },
  {
    id: 1027,
    name: 'Ethereum',
    symbol: 'ETH',
    slug: 'ethereum',
    num_market_pairs: 5901,
    date_added: '2015-08-07T00:00:00.000Z',
    tags: [
      'mineable',
      'pow',
      'paradigm-xzy-screener'
    ],
    max_supply: null,
    circulating_supply: 115925121.124,
    total_supply: 115925121.124,
    platform: null,
    cmc_rank: 2,
    last_updated: '2021-05-18T15:19:02.000Z',
    quote: { USD: [Object] }
  },
  {
    id: 1839,
    name: 'Binance Coin',
    symbol: 'BNB',
    slug: 'binance-coin',
    num_market_pairs: 596,
    date_added: '2017-07-25T00:00:00.000Z',
    tags: [
      'marketplace',
      'centralized-exchange',
      'payments',
      'binance-smart-chain',
      'alameda-research-portfolio',
      'multicoin-capital-portfolio'
    ],
    max_supply: 170532785,
    circulating_supply: 153432897,
    total_supply: 169432897,
    platform: null,
    cmc_rank: 3,
    last_updated: '2021-05-18T15:19:07.000Z',
    quote: { USD: [Object] }
  },
...
```
The first coin is from 2013, the second from 2015, third from 2017. None of these should have made it to the list because we are trying to fetch the most recent items. Are we making the api call in an inaccurate way or something ?

Reply:
let me check it again.

Reply:
@omeryousaf was it coming in correct order before when we were using "request-promise"?

Reply:
because I am using the same query parameters. I didn't change anything just replace the request-promise call with axios.

Reply:
@Shapokhan check kar k dekho na.